torch/jit.py

```diff
@@ -1,26 +1,375 @@
 import torch.autograd.function as function
 import torch._C
 from torch.autograd import Variable
-from torch.nn import Module
+from torch.nn import Module, ParameterList, Parameter
 from collections import defaultdict
 import itertools
 import types
 import contextlib
 import os
+import functools
 import torch.contrib._graph_vis as graph_vis
-# Example how to use:
-#
-# import torch.jit
-# model = model.RNNModel(args.model, ...)
-# model = torch.jit.traced(model)
 
 
-def flatten(x):
+HOLE = object()
+VOLATILE = object()
+
+
+# TODO: verify is not implemented yet
+def compile(arg=None, nderivs=1, params=tuple(), verify=False, optimize=True, enabled=True):
+    """
+    Mark a function or module as eligible for just-in-time compilation.  The
+    next time the function/module is executed, it is traced, and the trace is
+    compiled into an optimized representation which is run in lieu of the
+    original Python code upon subsequent invocations of the function/module.
+
+    The result of this function is stateful, so make sure you invoke it
+    only once per code you want to JIT compile.
+
+    .. note::
+
+        If your function/module takes non-Variable inputs, the JIT compiler
+        will compile a trace separately for each distinct input configuration.
```

Comment:
Explain what an "input configuration" is (i.e. Variables are compared by this and that, Python objects by this and that)

Reply:
The new copy is:

```
        A JIT compiled function/module may be compiled multiple times, as
        different inputs can result in different traces.  Currently, the
        JIT compiler conservatively assumes the trace may change if the
        `size` or `requires_grad` of `Variable` inputs change, or if
        any of the non-Variable inputs change.
```